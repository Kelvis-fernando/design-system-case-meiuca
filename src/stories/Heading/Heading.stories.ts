import type { Meta, StoryObj } from "@storybook/react";

import { Heading } from "./Heading";

const meta = {
  title: "Components/Heading",
  component: Heading,
  tags: ["autodocs"],
} satisfies Meta<typeof Heading>;

export default meta;
type Story = StoryObj<typeof meta>;

export const H1: Story = {
  args: {
    text: "First Heading",
    level: 1,
  },
};

export const H2: Story = {
  args: {
    text: "First Heading",
    level: 2,
  },
};

export const H3: Story = {
  args: {
    text: "First Heading",
    level: 3,
  },
};

export const H4: Story = {
  args: {
    text: "First Heading",
    level: 4,
  },
};

export const H5: Story = {
  args: {
    text: "First Heading",
    level: 5,
  },
};
export const H6: Story = {
  args: {
    text: "First Heading",
    level: 5,
  },
};
