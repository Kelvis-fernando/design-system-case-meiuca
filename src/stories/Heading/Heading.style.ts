import tokens from "../../../tokens/global.json";

export const headingStyle: React.CSSProperties = {
  color: tokens.neutral.color[5].value,
  fontFamily: tokens.font.family.highlight.value,
  margin: tokens.spacing.size.quarck.value,
};
