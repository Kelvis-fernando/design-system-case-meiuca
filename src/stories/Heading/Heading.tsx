import React from "react";
import { headingStyle } from "./Heading.style";

export interface HeadingProps {
  level: 1 | 2 | 3 | 4 | 5 | 6;
  text: string;
}

export const Heading: React.FC<HeadingProps> = ({ level, text }) => {
  const HeadingTag = `h${level}` as keyof JSX.IntrinsicElements;
  return <HeadingTag style={headingStyle}>{text}</HeadingTag>;
};
