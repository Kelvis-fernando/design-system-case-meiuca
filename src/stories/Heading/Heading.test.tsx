import { render, screen } from "@testing-library/react";
import { Heading } from "./Heading";

test("should render heading", () => {
  const { container } = render(<Heading level={2} text={"Heading"} />);
  const linkElement = screen.getByText(/heading/i);

  expect(linkElement).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
