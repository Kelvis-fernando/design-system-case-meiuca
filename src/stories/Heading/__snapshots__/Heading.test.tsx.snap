// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render heading 1`] = `
<div>
  <h2
    style="color: rgb(0, 0, 0); font-family: 'Archivo';"
  >
    Heading
  </h2>
</div>
`;
